browser/src/control/jsdialog/Widget.TreeView.js

```diff
@@ -426,15 +434,24 @@ function _handleKeyEvent(event, listElements, builder, data) {
 	if (event.key === 'ArrowDown') {
 		if (currIndex < 0)
 			_changeFocusedRow(listElements, currIndex, 0);
-		else if (currIndex < treeLength - 1)
-			_changeFocusedRow(listElements, currIndex, currIndex + 1);
-
+		else {
+			var nextIndex = currIndex + 1;
+			while (currIndex < treeLength - 1 && listElements[nextIndex].clientHeight <= 0)
+				nextIndex++;
+			if (nextIndex < treeLength)
+				_changeFocusedRow(listElements, currIndex, nextIndex);
+		}
 		preventDef = true;
 	} else if (event.key === 'ArrowUp') {
 		if (currIndex < 0)
 			_changeFocusedRow(listElements, currIndex, treeLength - 1);
-		else if (currIndex > 0)
-			_changeFocusedRow(listElements, currIndex, currIndex - 1);
+		else {
+			var nextIndex = currIndex - 1;
+			while (currIndex >= 0 && listElements[nextIndex].clientHeight <= 0)
```

Comment:
is that condition correct?
what it checks?

I would expect we want to check if `(nextIndex >= 0)` because we use it to access array listElements

Reply:
thx, you are right :)